### PR TITLE
Add spellchecking with dictionary

### PR DESCRIPTION
**Pull request by zachlatta**
This pull request improves upon @MaxWofford's original changes in #44. It
adds/changes the following:
- Custom dictionary support with a dictionary pre-populated from
  hackclub/hackclub
- Fixes the header link checking that was broken by the change to Alpine
- Strips stuff (like code blocks, link destinations, and HTML tags) we
  don't want to check for spelling before running the check itself
- Slightly clarifies the error message

I've tested this locally and it passes on `hackclub/hackclub` after the changes in hackclub/hackclub#629 are merged.

Replaces and closes #44 (along with closing #42).

### CODE REVIEW

**Comment by maxwofford** [comment]
@zachlatta This breaks link checking for me.

**Comment by maxwofford** [comment]
https://asciinema.org/a/97ixvw7k1addlgsjz2go0rkwq

**Comment by zachlatta** [comment]
Have you rebuilt the Docker image? I updated the Alpine Linux version from
3.2 to 3.3, which may fix that.

On Sun, Jan 3, 2016 at 2:21 AM, Max Wofford notifications@github.com
wrote:

> https://asciinema.org/a/97ixvw7k1addlgsjz2go0rkwq
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hackclub/markcop/pull/50#issuecomment-168485285.

## 

Zach Latta
http://zachlatta.com
https://github.com/zachlatta
https://twitter.com/zachlatta

File: lib/dictionary [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+AdVenture
+AIME
+AMC
+api
+API
+APIs
+Arduino
+bannings
+Bloomberg
+bogidon
```
Comment (maxwofford) [inline lib/dictionary:12]: Should this be lowercase?

File: lib/dictionary [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+flyer
+flyers
+FreeBSD
+freenode
+Fu
+fundraise
+Gaberiel
+GIF
+GIFs
+girlsWhoCode
```
Comment (maxwofford) [inline lib/dictionary:48]: Is this how they capitalize their name?

**Comment by maxwofford** [comment]
@zachlatta Thanks! Was on the wrong branch :/

Left a few comments in the dictionary, but other than that it's ready to merge. I'm going to merge for now and if any of the comments need to be addressed let's submit another PR.

File: lib/dictionary (hunk 2 above)
Comment (zachlatta) [inline lib/dictionary:48]: I believe it's the name of a club at @saescapa's high school, I got tripped up on this as well.

**Comment by zachlatta** [comment]
:heart:

File: lib/dictionary (hunk 2 above)
Comment (saescapa) [inline lib/dictionary:48]: I just looked it up, it's Girls Who Code.

Comment (zachlatta) [inline lib/dictionary:48]: Ah, cool. Just submitted #52 and hackclub/hackclub#632 to fix this :-).

On Sun, Jan 3, 2016 at 2:14 PM, Samuel Escapa notifications@github.com
wrote:

> In lib/dictionary
> https://github.com/hackclub/markcop/pull/50#discussion_r48697919:
> 
> > +decrement
> > +Django
> > +Dropbox
> > +EasyCTF
> > +EasyCTF's
> > +Flappy
> > +flyer
> > +flyers
> > +FreeBSD
> > +freenode
> > +Fu
> > +fundraise
> > +Gaberiel
> > +GIF
> > +GIFs
> > +girlsWhoCode
> 
> I just looked it up, it's Girls Who Code.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hackclub/markcop/pull/50/files#r48697919.

## 

Zach Latta
http://zachlatta.com
https://github.com/zachlatta
https://twitter.com/zachlatta

**Pull request merged**